code/ORM/FieldType/JSONText.php

```diff
@@ -250,15 +250,20 @@ public function toSSTypes(array $data)
      */
     public function toArray($value = null)
     {
+        $output = [];
         $value = $value ?: $this->getValue();
-        $decoded = json_decode($value, true);
 
-        if (is_null($decoded)) {
-            $msg = 'Decoded JSON is invalid.';
-            throw new JSONTextDataException($msg);
+        // Check for value
+        if (!empty($value)) {
+            $output = json_decode($this->getValue(), true);
+
+            // Check for parse errors
+            if (JSON_ERROR_NONE !== json_last_error()) {
```

Comment:
What's the advantage of this approach, over checking for `null` returned from a bad `json_decode()` call? (Where "bad" can also mean, checking for too deep-a-depth of JSON object)

Reply:
Specificity. I figured while we're hear, we might as well differentiate.